backend/src/org/commcare/xml/ProfileParser.java

```diff
@@ -109,113 +149,78 @@ public Profile parse() throws InvalidStructureException, IOException, XmlPullPar
-                        getNextTagInBlock("suite");
-                        Resource resource = new ResourceParser(parser, maximumResourceAuthority).parse();
+    private void parseLogin() throws InvalidStructureException, IOException, XmlPullParserException{
+        // Get the resource block or fail out
+        getNextTagInBlock("login");
+        Resource resource = new ResourceParser(parser, maximumResourceAuthority).parse();
+        table.addResource(resource, table.getInstallers().getLoginImageInstaller(), resourceId, initialResourceStatus);
+    }
+
+    private void parseFeatures(Profile profile) throws XmlPullParserException, IOException, InvalidStructureException {
+        while (nextTagInBlock("features")) {
+            String tag = parser.getName().toLowerCase();
+            String active = parser.getAttributeValue(null, "active");
+            boolean isActive = false;
+            if (active != null && active.toLowerCase().equals("true")) {
+                isActive = true;
+            }
+            if (tag.equals("checkoff")) {
 
-                        //TODO: Possibly add a real parent reference if we decide these go in the table
-                        table.addResource(resource, table.getInstallers().getSuiteInstaller(), resourceId, initialResourceStatus);
+            } else if (tag.equals("reminders")) {
+                if (nextTagInBlock("reminders")) {
+                    checkNode("time");
+                    String reminderTime = parser.nextText();
+                }
+            } else if (tag.equals("package")) {
+                //nothing (yet)
+            } else if (tag.equals("users")) {
+                while (nextTagInBlock("users")) {
+                    if (parser.getName().toLowerCase().equals("registration")) {
+                        profile.addPropertySetter("user_reg_namespace", parser.nextText(), true);
+                    } else if (parser.getName().toLowerCase().equals("logo")) {
+                        String logo = parser.nextText();
+                        profile.addPropertySetter("cc_login_image", logo, true);
                     } else {
-                        System.out.println("Unrecognized Tag: "
-                                + parser.getName());
+                        throw new InvalidStructureException("Unrecognized tag " + parser.getName() + " inside of users feature block", parser);
                     }
-                } else if (eventType == KXmlParser.END_TAG) {
-                    // we shouldn't ever get this I don't believe, maybe on the
-                    // last node?
-                } else if (eventType == KXmlParser.TEXT) {
-                    // Shouldn't ever get this (Delete the if, if so).
                 }
-                eventType = parser.next();
-            } while (eventType != KXmlParser.END_DOCUMENT);
-
-            return profile;
+            } else if (tag.equals("sense")) {

```

Comment:
I refrained from doing so in case that was in there to remind someone to implement something at some point.. very happy for it to be gone if that's not the case.